cli/azd/pkg/graphsdk/applications_models.go

```diff
@@ -0,0 +1,44 @@
+package graphsdk
+
+import "time"
+
+// A Microsoft Graph Application entity.
+type Application struct {
+	Id                  *string                          `json:"id"`
+	AppId               *string                          `json:"appId"`
+	DisplayName         string                           `json:"displayName"`
+	Description         *string                          `json:"description"`
+	PasswordCredentials []*ApplicationPasswordCredential `json:"passwordCredentials"`
+}
+
+type ApplicationCreateRequest struct {
+	Application
+}
+
+// A list of applications returned from the Microsoft Graph.
+type ApplicationListResponse struct {
+	NextLink *string       `json:"@odata.nextLink"`
```

Comment:
Do we need to do something long term to handle pageable responses?  A quick Control-F makes it seem like we never use `NextLink`, and so I'm wondering if there are multiple pages in the response and the item we are looking for is in the second page, will we miss it?

Reply:
The `pager` is already in our code (*runtime.Pager).  You need to implement a pager for the application `get` response.

See an example of implementing the pager at: pkg/azdo/paged_response_api.go

The response from the server can return n applications in a page, where `n` can be >=0.  So, we must have the pager

Reply:
I was going to implement it all the the responses where we get a list of things are from the result of a filter call like `$filter=displayName eq '{some_value}'`

This basically will give us a zero of 1 use case so I'm going to hold off on implementing paging support right now.

Reply:
for `azd pipeline config`, it would be ideal if we could list the existing `applications` available.
Similar to selecting a subscription or location, we can let the user to pick a service principal from a list. Like:
- Choose an existing service principal or create a new one?
  - Create new
  - Foo
  - Bar

Reply:
Great idea, but out of scope for this PR.  Logged new issue #972 to track this.